### PR TITLE
WT-12872 Make the pinned_ratio check in test_chunkcache03 more lenient

### PR DESCRIPTION
**Pull request by ajmorton**
Occasionally the float division in `self.assertEqual(total_chunks/pinned_chunks, 2)` would evaluate to 2.0625 instead of 2 and fail the test. This assertion works on the assumption that all tables in the test have been populated to be exactly the same size and use the exact same number of chunks, but there is no guarantee that this is true. Instead round the divided result to handle small variation is number of chunks used by each table.

This PR also updates the check to test a few more combinations of pinned tables for certainty.

### CODE REVIEW

**Review by korteland — APPROVED** [review]
LGTM - this is a nice improvement, thank you!

File: test/suite/test_chunkcache03.py [hunk 1]
```diff
@@ -119,11 +119,27 @@ def test_chunkcache03(self):
...
-        # Assert that pinned chunks make up 50% of the total chunks.
-        self.assertEqual(total_chunks/pinned_chunks, 2)
+        # Assert that pinning a table also pins its backing chunks.
+        # We can't measure this directly, so exploit the fact that all four tables have been
+        # populated to roughly the same size and will use ~25% of available chunks each.
 
-        # Reconfigure wiredtiger and mark the pinned objects as unpinned and vice-versa.
-        self.conn.reconfigure('chunk_cache=[pinned=("table:chunkcache03", "table:chunkcache04")]')
+        # We start with two tables pinned. This is half the total chunks
+        self.assertEqual(round(total_chunks/pinned_chunks), 2)
```
Comment (marcbutler) [inline test/suite/test_chunkcache03.py:127]: As there are no digits of precision, you could use the floor division operator:
```suggestion
        self.assertEqual(total_chunks // pinned_chunks, 2)
```
Just a fyi: personally I think both are fine.

**Review by Siddhartha8899 — APPROVED** [review]

File: test/suite/test_chunkcache03.py (hunk 1 above)
Comment (ajmorton) [reply test/suite/test_chunkcache03.py:127]: I considered it initially, but while we haven't seen it happen I wanted to protect against values like 1.9875 as well.

Comment (marcbutler) [reply test/suite/test_chunkcache03.py:127]: Thank you for clarifying.

**Pull request merged**